pkg/provider/azure_vmssflex.go

```diff
@@ -926,17 +926,7 @@ func (fs *FlexScaleSet) EnsureBackendPoolDeleted(service *v1.Service, backendPoo
 		}
 	}
 
-	// 1. Ensure the backendPoolID is deleted from the VMSS.
```

Comment:
Since we ensure all vmss are decoupled from the lb in azure_vmss.go, we don't need it here, right?

Reply:
I think we should still need it. Per AKS request, I will add another vmType == vmssflex for pure vmssflex cluster. In this case, we skip the node type check and only initialize FlexScaleSet ( similar to the pure standalone VM cluster).